src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Type.hs

```diff
@@ -154,13 +155,9 @@ data ConsolidatedRowReferences b a = ConsolidatedRowReferences
   , theRowWidth :: RowWidth
   }
 
--- | This wrapper facilitates naming the original structure
--- (i.e. the "payload" for recognition)
--- for the purpose of both UI display and internal uniqueness,
--- while remaining agnostic to its internals.
-data NamedOriginal b = NamedOriginal
-  { getName :: StructureName
-  , orig :: NamedGrid b
```

Comment:
Don't need `NamedOriginal` anymore; we just access the `name` through the `NamedArea` record.